gateway/tenant.go

```diff
@@ -0,0 +1,62 @@
+package gateway
+
+import (
+	"log"
+	"os"
+
+	"gopkg.in/yaml.v2"
+)
+
+type Tenant struct {
+	ID       string `yaml:"ID"`
+	Username string `yaml:"username"`
+	Password string `yaml:"password"`
+}
+
+type config struct {
+	Basic map[string]Tenant `yaml:"basic"`
+}
+
+var tenants = make([]Tenant, 0)
+var isTenantsInitialized = false
```

Comment:
Don't use global variables, you can put these in main method if you don't have a better place